Corrupted output from code formatting

### Server Version

```
$ terraform-ls --version
0.25.2
```

### Terraform Version

```
$ terraform --version
Terraform v1.1.6
on linux_amd64
```

### Client Version

```
$ nvim --version
NVIM v0.6.1
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by team+vim@tracker.debian.org
```

### Terraform Configuration Files

**`testcase.tf`**

```hcl
resource "aws_lambda_function" "f" {
    environment {
        variables = {
            a = "b"
        }
    }
}
```

### Log Output

<details>
<summary>Log file</summary>

```
2022/02/20 14:24:58 serve_command.go:162: Starting terraform-ls 0.25.2
2022/02/20 14:24:58 service.go:90: Preparing new session ...
2022/02/20 14:24:58 langserver.go:94: Starting server (pid 53; concurrency: 4) ...
2022/02/20 14:24:58 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:24:58 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:24:58 rpc_logger.go:29: Incoming request for "initialize" (ID 1): {"trace":"off","workspaceFolders":[{"uri":"file:\/\/\/d\/testcase","name":"\/d\/testcase"}],"capabilities":{"window":{"workDoneProgress":true,"showDocument":{"support":false},"showMessage":{"messageActionItem":{"additionalPropertiesSupport":false}}},"callHierarchy":{"dynamicRegistration":false},"textDocument":{"references":{"dynamicRegistration":false},"publishDiagnostics":{"relatedInformation":true,"tagSupport":{"valueSet":[1,2]}},"typeDefinition":{"linkSupport":true},"rename":{"dynamicRegistration":false,"prepareSupport":true},"declaration":{"linkSupport":true},"codeAction":{"dataSupport":true,"resolveSupport":{"properties":["edit"]},"dynamicRegistration":false,"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","Empty","QuickFix","Refactor","RefactorExtract","RefactorInline","RefactorRewrite","Source","SourceOrganizeImports","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"contextSupport":false,"completionItem":{"insertReplaceSupport":true,"deprecatedSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"commitCharactersSupport":true,"tagSupport":{"valueSet":[1]},"labelDetailsSupport":true,"preselectSupport":true,"snippetSupport":true,"documentationFormat":["markdown","plaintext"]},"dynamicRegistration":false,"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"signatureHelp":{"dynamicRegistration":false,"signatureInformation":{"parameterInformation":{"labelOffsetSupport":true},"activeParameterSupport":true,"documentationFormat":["markdown","plaintext"]}},"documentHighlight":{"dynamicRegistration":false},"definition":{"linkSupport":true},"documentSymbol":{"hierarchicalDocumentSymbolSupport":true,"dynamicRegistration":false,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"hover":{"dynamicRegistration":false,"contentFormat":["markdown","plaintext"]},"implementation":{"linkSupport":true},"synchronization":{"didSave":true,"willSave":false,"dynamicRegistration":false,"willSaveWaitUntil":false}},"workspace":{"workspaceFolders":true,"symbol":{"hierarchicalWorkspaceSymbolSupport":true,"dynamicRegistration":false,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"workspaceEdit":{"resourceOperations":["rename","create","delete"]},"applyEdit":true,"configuration":true}},"processId":51,"initializationOptions":{},"rootUri":"file:\/\/\/d\/testcase","clientInfo":{"name":"Neovim","version":"0.6.1"},"rootPath":"\/d\/testcase"}
2022/02/20 14:25:01 watcher.go:283: watching for changes ...
2022/02/20 14:25:01 walker.go:191: asynchronously walking through /d/testcase
2022/02/20 14:25:01 walker.go:240: skipping /d/testcase/.git
2022/02/20 14:25:01 rpc_logger.go:50: Response to "initialize" (ID 1): {"capabilities":{"textDocumentSync":{"openClose":true,"change":2,"save":{}},"completionProvider":{"triggerCharacters":[".","["],"completionItem":{}},"hoverProvider":true,"signatureHelpProvider":{},"declarationProvider":{},"definitionProvider":true,"referencesProvider":true,"documentSymbolProvider":true,"codeActionProvider":{"codeActionKinds":["source.formatAll.terraform"]},"codeLensProvider":{},"documentLinkProvider":{},"workspaceSymbolProvider":true,"documentFormattingProvider":true,"documentOnTypeFormattingProvider":{"firstTriggerCharacter":""},"executeCommandProvider":{"commands":["terraform-ls.module.callers","terraform-ls.module.calls","terraform-ls.module.providers","terraform-ls.rootmodules","terraform-ls.terraform.init","terraform-ls.terraform.validate"],"workDoneProgress":true},"semanticTokensProvider":{"legend":{"tokenTypes":[],"tokenModifiers":[]},"full":false},"workspace":{"workspaceFolders":{"supported":true,"changeNotifications":"workspace/didChangeWorkspaceFolders"}}},"serverInfo":{"name":"terraform-ls","version":"0.25.2"}}
2022/02/20 14:25:01 opts.go:254: Completed 1 requests [2.995581692s elapsed]
2022/02/20 14:25:01 walker.go:321: walking of /d/testcase finished
2022/02/20 14:25:01 walker.go:197: async walking through /d/testcase finished
2022/02/20 14:25:01 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:25:01 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:25:01 rpc_logger.go:29: Incoming notification for "initialized": {}
2022/02/20 14:25:01 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:25:01 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:25:01 rpc_logger.go:29: Incoming notification for "textDocument/didOpen": {"textDocument":{"uri":"file:\/\/\/d\/testcase\/testcase.tf","languageId":"terraform","version":0,"text":"resource \"aws_lambda_function\" \"f\" {\n    environment {\n        variables = {\n            a = \"b\"\n        }\n    }\n}\n"}}
2022/02/20 14:25:01 module_manager.go:66: MM: adding new module: /d/testcase
2022/02/20 14:25:01 did_open.go:44: opened module: /d/testcase
2022/02/20 14:25:01 module_loader.go:185: ML: enqueing "OpTypeParseModuleConfiguration" module operation: "/d/testcase"
2022/02/20 14:25:01 module_loader.go:185: ML: enqueing "OpTypeParseVariables" module operation: "/d/testcase"
2022/02/20 14:25:01 module_loader.go:185: ML: enqueing "OpTypeLoadModuleMetadata" module operation: "/d/testcase"
2022/02/20 14:25:01 module_loader.go:120: ML: executing "OpTypeParseModuleConfiguration" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:120: ML: executing "OpTypeParseVariables" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:185: ML: enqueing "OpTypeDecodeReferenceTargets" module operation: "/d/testcase"
2022/02/20 14:25:01 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:25:01 module_loader.go:185: ML: enqueing "OpTypeDecodeReferenceOrigins" module operation: "/d/testcase"
2022/02/20 14:25:01 module_loader.go:172: ML: finished "OpTypeParseVariables" for "/d/testcase"
2022/02/20 14:25:01 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:25:01 opts.go:254: Request check error for "rust-analyzer/inlayHints" (params "{\"textDocument\":{\"uri\":\"file:\\/\\/\\/d\\/testcase\\/testcase.tf\"}}"): [-32601] no such method "rust-analyzer/inlayHints"
2022/02/20 14:25:01 module_loader.go:185: ML: enqueing "OpTypeGetTerraformVersion" module operation: "/d/testcase"
2022/02/20 14:25:01 module_loader.go:172: ML: finished "OpTypeParseModuleConfiguration" for "/d/testcase"
2022/02/20 14:25:01 watcher.go:75: adding module for watching: /d/testcase
2022/02/20 14:25:01 module_loader.go:120: ML: executing "OpTypeDecodeReferenceTargets" for "/d/testcase"
2022/02/20 14:25:01 rpc_logger.go:45: Error for "rust-analyzer/inlayHints" (ID 2): [-32601] no such method "rust-analyzer/inlayHints"
2022/02/20 14:25:01 opts.go:254: Completed 1 requests [101.02193ms elapsed]
2022/02/20 14:25:01 module_loader.go:120: ML: executing "OpTypeDecodeReferenceOrigins" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:172: ML: finished "OpTypeDecodeReferenceTargets" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:172: ML: finished "OpTypeDecodeReferenceOrigins" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:120: ML: executing "OpTypeLoadModuleMetadata" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:120: ML: executing "OpTypeGetTerraformVersion" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:172: ML: finished "OpTypeLoadModuleMetadata" for "/d/testcase"
2022/02/20 14:25:01 module_loader.go:172: ML: finished "OpTypeGetTerraformVersion" for "/d/testcase"
2022/02/20 14:25:10 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:25:10 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:25:10 rpc_logger.go:29: Incoming request for "textDocument/formatting" (ID 3): {"options":{"tabSize":8,"insertSpaces":false},"textDocument":{"uri":"file:\/\/\/d\/testcase\/testcase.tf"}}
2022/02/20 14:25:10 formatting.go:41: formatting document via "/usr/bin/terraform"
2022/02/20 14:25:10 rpc_logger.go:50: Response to "textDocument/formatting" (ID 3): [{"range":{"start":{"line":1,"character":0},"end":{"line":5,"character":0}},"newText":"  environment {\n    variables = {\n      a = \"b\"\n"},{"range":{"start":{"line":6,"character":0},"end":{"line":7,"character":0}},"newText":"  }\n"}]
2022/02/20 14:25:10 opts.go:254: Completed 1 requests [47.253852ms elapsed]
2022/02/20 14:25:10 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:25:10 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:25:10 rpc_logger.go:29: Incoming notification for "textDocument/didChange": {"contentChanges":[{"rangeLength":1,"range":{"start":{"line":6,"character":0},"end":{"line":6,"character":1}},"text":"  }\n"},{"rangeLength":73,"range":{"start":{"line":1,"character":2},"end":{"line":5,"character":5}},"text":"environment {\n    variables = {\n      a = \"b\"\n    }"},{"rangeLength":1,"range":{"start":{"line":6,"character":0},"end":{"line":7,"character":0}},"text":""}],"textDocument":{"uri":"file:\/\/\/d\/testcase\/testcase.tf","version":6}}
2022/02/20 14:25:10 module_loader.go:185: ML: enqueing "OpTypeParseModuleConfiguration" module operation: "/d/testcase"
2022/02/20 14:25:10 module_loader.go:185: ML: enqueing "OpTypeParseVariables" module operation: "/d/testcase"
2022/02/20 14:25:10 module_loader.go:185: ML: enqueing "OpTypeLoadModuleMetadata" module operation: "/d/testcase"
2022/02/20 14:25:10 module_loader.go:120: ML: executing "OpTypeParseModuleConfiguration" for "/d/testcase"
2022/02/20 14:25:10 module_loader.go:185: ML: enqueing "OpTypeDecodeReferenceTargets" module operation: "/d/testcase"
2022/02/20 14:25:10 module_loader.go:185: ML: enqueing "OpTypeDecodeReferenceOrigins" module operation: "/d/testcase"
2022/02/20 14:25:10 module_loader.go:120: ML: executing "OpTypeParseVariables" for "/d/testcase"
2022/02/20 14:25:10 module_loader.go:172: ML: finished "OpTypeParseModuleConfiguration" for "/d/testcase"
2022/02/20 14:25:10 module_loader.go:172: ML: finished "OpTypeParseVariables" for "/d/testcase"
2022/02/20 14:25:10 opts.go:254: Posting server notification "textDocument/publishDiagnostics" {"uri":"file:///d/testcase","diagnostics":[]}
2022/02/20 14:25:10 opts.go:254: Posting server notification "textDocument/publishDiagnostics" {"uri":"file:///d/testcase/testcase.tf","diagnostics":[{"range":{"start":{"line":0,"character":35},"end":{"line":0,"character":36}},"severity":1,"source":"HCL","message":"Unclosed configuration block: There is no closing brace for this block before the end of the file. This may be caused by incorrect brace nesting elsewhere in this file."}]}
2022/02/20 14:25:10 opts.go:254: Posting server notification "textDocument/publishDiagnostics" {"uri":"file:///d/testcase","diagnostics":[]}
2022/02/20 14:25:10 opts.go:254: Posting server notification "textDocument/publishDiagnostics" {"uri":"file:///d/testcase/testcase.tf","diagnostics":[{"range":{"start":{"line":0,"character":35},"end":{"line":0,"character":36}},"severity":1,"source":"HCL","message":"Unclosed configuration block: There is no closing brace for this block before the end of the file. This may be caused by incorrect brace nesting elsewhere in this file."}]}
2022/02/20 14:25:10 module_loader.go:120: ML: executing "OpTypeDecodeReferenceTargets" for "/d/testcase"
2022/02/20 14:25:10 module_loader.go:120: ML: executing "OpTypeDecodeReferenceOrigins" for "/d/testcase"
2022/02/20 14:25:10 provider_schema.go:196: PSS: getting provider schema (/d/testcase, registry.terraform.io/-/aws, )
2022/02/20 14:25:10 provider_schema.go:196: PSS: getting provider schema (/d/testcase, registry.terraform.io/-/aws, )
2022/02/20 14:25:10 module_loader.go:172: ML: finished "OpTypeDecodeReferenceOrigins" for "/d/testcase"
2022/02/20 14:25:10 module_loader.go:172: ML: finished "OpTypeDecodeReferenceTargets" for "/d/testcase"
2022/02/20 14:25:10 module_loader.go:120: ML: executing "OpTypeLoadModuleMetadata" for "/d/testcase"
2022/02/20 14:25:10 module_loader.go:172: ML: finished "OpTypeLoadModuleMetadata" for "/d/testcase"
2022/02/20 14:25:10 opts.go:254: Posting server notification "textDocument/publishDiagnostics" {"uri":"file:///d/testcase/testcase.tf","diagnostics":[{"range":{"start":{"line":0,"character":35},"end":{"line":0,"character":36}},"severity":1,"source":"HCL","message":"Unclosed configuration block: There is no closing brace for this block before the end of the file. This may be caused by incorrect brace nesting elsewhere in this file."}]}
2022/02/20 14:25:10 opts.go:254: Posting server notification "textDocument/publishDiagnostics" {"uri":"file:///d/testcase","diagnostics":[]}
2022/02/20 14:25:12 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:25:12 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:25:12 rpc_logger.go:29: Incoming request for "shutdown" (ID 4): 
2022/02/20 14:25:12 service.go:513: stopping walker for session ...
2022/02/20 14:25:12 walker.go:103: stopping walker
2022/02/20 14:25:12 service.go:515: walker stopped
2022/02/20 14:25:12 service.go:519: stopping watcher for session ...
2022/02/20 14:25:12 service.go:524: watcher stopped
2022/02/20 14:25:12 service.go:529: cancelling any module loading ...
2022/02/20 14:25:12 service.go:531: module loading cancelled
2022/02/20 14:25:12 module_loader.go:61: Cancelling module loader...
2022/02/20 14:25:12 rpc_logger.go:50: Response to "shutdown" (ID 4): null
2022/02/20 14:25:12 opts.go:254: Completed 1 requests [14.417358ms elapsed]
2022/02/20 14:25:12 opts.go:254: Received request batch of size 1 (qlen=0)
2022/02/20 14:25:12 opts.go:254: Dequeued request batch of length 1 (qlen=0)
2022/02/20 14:25:12 rpc_logger.go:29: Incoming notification for "exit": 
2022/02/20 14:25:12 opts.go:254: Server signaled to stop with err=EOF
2022/02/20 14:25:12 opts.go:254: Error reading from client: EOF
2022/02/20 14:25:12 service.go:504: session stopped unexpectedly (err: <nil>)
2022/02/20 14:25:12 service.go:513: stopping walker for session ...
2022/02/20 14:25:12 service.go:515: walker stopped
2022/02/20 14:25:12 service.go:519: stopping watcher for session ...
2022/02/20 14:25:12 service.go:524: watcher stopped
2022/02/20 14:25:12 service.go:529: cancelling any module loading ...
2022/02/20 14:25:12 service.go:531: module loading cancelled
2022/02/20 14:25:12 langserver.go:107: Stopping server (pid 53) ...
2022/02/20 14:25:12 langserver.go:111: Server (pid 53) stopped.
```

</details>

### Expected Behavior

Running the LSP code formatter on the current buffer using `:lua vim.lsp.buf.formatting()` should not add syntax errors.

### Actual Behavior

The code formatter removes a closing brace, causing a syntax error:

```hcl
resource "aws_lambda_function" "f" {    
  environment {    
    variables = {    
      a = "b"        
    }    
  }      
```

Error text: `Unclosed configuration block: There is no closing brace for this block before the end of the file.`

### Steps to Reproduce

1. Create `testcase.tf` as above
2. Open in nvim 0.6.1 with nvim-lspconfig
3. Run `:lua vim.lsp.buf.formatting()`

### Additional info

When running `terraform fmt` on the same file, the output seems to be correct. I couldn't reproduce the problem using `terraform fmt`.

Here is the JSONRPC response to the formatting request:

```json
{
  "jsonrpc": "2.0",
  "id": 3,
  "result": [
    {
      "range": {
        "start": {
          "line": 1,
          "character": 0
        },
        "end": {
          "line": 5,
          "character": 0
        }
      },
      "newText": "  environment {\n    variables = {\n      a = \"b\"\n"
    },
    {
      "range": {
        "start": {
          "line": 6,
          "character": 0
        },
        "end": {
          "line": 7,
          "character": 0
        }
      },
      "newText": "  }\n"
    }
  ]
}
```

--- a/internal/hcl/diff.go
+++ b/internal/hcl/diff.go
@@ -62,12 +62,14 @@ func diffLines(filename string, beforeLines, afterLines source.Lines) document.C
 
 	for _, group := range m.GetGroupedOpCodes(context) {
 		for _, c := range group {
-			beforeStart, beforeEnd := c.I1, c.I2
-			afterStart, afterEnd := c.J1, c.J2
-
 			if c.Tag == OpEqual {
 				continue
 			}
+
+			// lines to pick from the original document (to delete/replace/insert to)
+			beforeStart, beforeEnd := c.I1, c.I2
+			// lines to pick from the new document (to replace ^ with)
+			afterStart, afterEnd := c.J1, c.J2
 
 			if c.Tag == OpReplace {
 				var rng *hcl.Range
@@ -121,6 +123,10 @@ func diffLines(filename string, beforeLines, afterLines source.Lines) document.C
 				if beforeStart == beforeEnd {
 					line := beforeLines[beforeStart]
 					insertRng = line.Range.Ptr()
+
+					// We're inserting to the beginning of the line
+					// which we represent as 0-length range in HCL
+					insertRng.End = insertRng.Start
 				} else {
 					for i, line := range beforeLines[beforeStart:beforeEnd] {
 						if i == 0 {

--- a/internal/hcl/diff_test.go
+++ b/internal/hcl/diff_test.go
@@ -179,6 +179,43 @@ ccc`,
 			},
 		},
 		{
+			name: "insertion to existing line",
+			beforeCfg: `resource "aws_lambda_function" "f" {
+    environment {
+        variables = {
+            a = "b"
+        }
+    }
+}
+`,
+			afterCfg: `resource "aws_lambda_function" "f" {
+  environment {
+    variables = {
+      a = "b"
+    }
+  }
+}
+`,
+			expectedChanges: document.Changes{
+				&fileChange{
+					newText: "  environment {\n    variables = {\n      a = \"b\"\n",
+					rng: &hcl.Range{
+						Filename: "test.tf",
+						Start:    hcl.Pos{Line: 2, Column: 1, Byte: 37},
+						End:      hcl.Pos{Line: 6, Column: 1, Byte: 107},
+					},
+				},
+				&fileChange{
+					newText: "  }\n",
+					rng: &hcl.Range{
+						Filename: "test.tf",
+						Start:    hcl.Pos{Line: 7, Column: 1, Byte: 113},
+						End:      hcl.Pos{Line: 7, Column: 1, Byte: 113},
+					},
+				},
+			},
+		},
+		{
 			"empty to newline",
 			``,
 			`

--- a/internal/langserver/handlers/formatting_test.go
+++ b/internal/langserver/handlers/formatting_test.go
@@ -277,3 +277,126 @@ func TestLangServer_formatting_variables(t *testing.T) {
 			]
 		}`)
 }
+
+func TestLangServer_formatting_diffBug(t *testing.T) {
+	tmpDir := TempDir(t)
+
+	cfg := `resource "aws_lambda_function" "f" {
+    environment {
+        variables = {
+            a = "b"
+        }
+    }
+}
+`
+	formattedCfg := `resource "aws_lambda_function" "f" {
+  environment {
+    variables = {
+      a = "b"
+    }
+  }
+}
+`
+
+	ls := langserver.NewLangServerMock(t, NewMockSession(&MockSessionInput{
+		TerraformCalls: &exec.TerraformMockCalls{
+			PerWorkDir: map[string][]*mock.Call{
+				tmpDir.Path(): {
+					{
+						Method:        "Version",
+						Repeatability: 1,
+						Arguments: []interface{}{
+							mock.AnythingOfType(""),
+						},
+						ReturnArguments: []interface{}{
+							version.Must(version.NewVersion("0.12.0")),
+							nil,
+							nil,
+						},
+					},
+					{
+						Method:        "GetExecPath",
+						Repeatability: 1,
+						ReturnArguments: []interface{}{
+							"",
+						},
+					},
+					{
+						Method:        "Format",
+						Repeatability: 1,
+						Arguments: []interface{}{
+							mock.AnythingOfType("*context.valueCtx"),
+							[]byte(cfg),
+						},
+						ReturnArguments: []interface{}{
+							[]byte(formattedCfg),
+							nil,
+						},
+					},
+				},
+			},
+		},
+	}))
+	stop := ls.Start(t)
+	defer stop()
+
+	ls.Call(t, &langserver.CallRequest{
+		Method: "initialize",
+		ReqParams: fmt.Sprintf(`{
+	    "capabilities": {},
+	    "rootUri": %q,
+	    "processId": 12345
+	}`, tmpDir.URI)})
+	ls.Notify(t, &langserver.CallRequest{
+		Method:    "initialized",
+		ReqParams: "{}",
+	})
+	ls.Call(t, &langserver.CallRequest{
+		Method: "textDocument/didOpen",
+		ReqParams: fmt.Sprintf(`{
+		"textDocument": {
+			"version": 0,
+			"languageId": "terraform",
+			"text": `+fmt.Sprintf("%q", cfg)+`,
+			"uri": "%s/main.tf"
+		}
+	}`, tmpDir.URI)})
+	ls.CallAndExpectResponse(t, &langserver.CallRequest{
+		Method: "textDocument/formatting",
+		ReqParams: fmt.Sprintf(`{
+			"textDocument": {
+				"uri": "%s/main.tf"
+			}
+		}`, tmpDir.URI)}, `{
+			"jsonrpc": "2.0",
+			"id": 3,
+			"result": [
+				{
+					"range": {
+						"start": {
+							"line": 1,
+							"character": 0
+						},
+						"end": {
+							"line": 5,
+							"character": 0
+						}
+					},
+					"newText": "  environment {\n    variables = {\n      a = \"b\"\n"
+				},
+				{
+					"range": {
+						"start": {
+							"line": 6,
+							"character": 0
+						},
+						"end": {
+							"line": 6,
+							"character": 0
+						}
+					},
+					"newText":"  }\n"
+				}
+			]
+		}`)
+}

